### PR TITLE
TF EKS example draft resources

### PR DESCRIPTION
**Pull request by klizhentas**
This is for https://github.com/gravitational/teleport/issues/2829

Please note that we have to give folks end to end guide from scratch, you should also review and patch some missing pieces marked with `<<>>`` as variables. @webvictim can help with cleanup of terraform.

Also ping me if you have any questions.

### CODE REVIEW

**Comment by benarent** [comment]
👍 for providing an easy TF scripts for the different usage scenarios.  

 Should we work on creating these into [Terraform Modules](https://www.terraform.io/docs/configuration/modules.html) vs just examples scripts? It would make these easier to go from POC Example -> Reusable Production Terraform.

**Comment by klizhentas** [comment]
I like the idea of modules, yes, having configurable modules highlighting parameters would be nice, however one thing to keep in mind that lots of times folks will be copy pasting snippets as well, so they should be copy-pasteable if that's a term :)

File: examples/eks/eks.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  default = "172.31.0.0/16"
+}
+
+variable "cluster-name" {
+  default = "terraform-eks-demo"
+  type    = "string"
+}
+
+provider "aws" {
+  version    = "~> 2.1.0"
```
Comment (benarent) [inline examples/eks/eks.tf]: I would recommend using 2.7.0 as this is compatible with Terraform 0.12.0. There is a fair amount of improvements in TF 0.12.0 https://www.terraform.io/upgrade-guides/0-12.html

File: examples/eks/worker_asg.tf [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+set -o xtrace
+/etc/eks/bootstrap.sh --apiserver-endpoint '${aws_eks_cluster.demo.endpoint}' --b64-cluster-ca '${aws_eks_cluster.demo.certificate_authority.0.data}' '${var.cluster-name}'
+USERDATA
+}
+
+resource "aws_launch_configuration" "demo" {
+  associate_public_ip_address = true
+  iam_instance_profile        = "${aws_iam_instance_profile.demo-node.name}"
+  image_id                    = "${data.aws_ami.eks-worker.id}"
+  instance_type               = "m4.large"
```
Comment (benarent) [inline examples/eks/worker_asg.tf]: Is this the smallest possible instance? Or is it possible to run on a smaller machine? Just wanted to outline expected costs for customers trying this out.

**Comment by benarent** [comment]
I explored this a little more, and to help add this docs. I started by creating this diagram. Let me know if the gist of it is right?  I wasn't 100% clear on how EKS workers work, but wanted to see if I've got the right jist here. 

image

**Pull request closed without merging**